### PR TITLE
[WIP] APIv4: allow a field named "is_active" to default to false

### PR DESCRIPTION
**Pull request by semseysandor**
Overview
----------------------------------------
If an entity has a field named "is_active" and it's default is defined as `0` in the schema, APIv4 overrides it to `1`.
This effectively prevents these entities to be created as not active by default (when "is_active" is omitted from parameters).

Before
----------------------------------------
(Output trimmed for brevity)
```
$ cv api4 Dashboard.create
[
    {
        "id": 17,
        "is_active": true,
    }
]
```

After
----------------------------------------
(Output trimmed for brevity)
```
$ cv api4 Dashboard.create
[
    {
        "id": 18,
    }
]
$ cv api4 Dashboard.get +w 'id = 18'
[
    {
        "id": 18,
        "is_active": false,
    }
]

```

Technical Details
----------------------------------------
Similar to #27302 

Comments
----------------------------------------
I ran into this in an extension, so it's possible to rename the field to "is_enabled" or something like that to workaround this limitation.
But there is one example in core: Dashboard entity, where APIv4 don't follow specifications, so this should be fixed.

### CODE REVIEW

File: Civi/Api4/Service/Spec/SpecGatherer.php [hunk 1]
```diff
@@ -97,7 +97,7 @@ private function addDAOFields(string $entityName, string $action, RequestSpec $s
       if (!empty($DAOField['component']) && !\CRM_Core_Component::isEnabled($DAOField['component'])) {
         continue;
       }
-      if ($DAOField['name'] == 'is_active' && empty($DAOField['default'])) {
+      if ($DAOField['name'] == 'is_active' && (!isset($DAOField['default']) || $DAOField['default'] === '')) {
         $DAOField['default'] = '1';
       }
```
Comment (colemanw) [inline Civi/Api4/Service/Spec/SpecGatherer.php:102]: Oof, I forgot this code existed. I think we should just delete it. Our schema used to be very inconsistent but there's been a lot of work lately to clean up boolean columns and ensure they always have the correct default set at the sql level so I think we can rely on that now and get rid of this.

Comment (semseysandor) [reply Civi/Api4/Service/Spec/SpecGatherer.php:102]: OK. What about the test?
It seems a bit redundant after deletion, but more test is usually better than less :smiley:

Comment (colemanw) [reply Civi/Api4/Service/Spec/SpecGatherer.php:102]: Yes keep it. Always good to have a test!

Comment (semseysandor) [reply Civi/Api4/Service/Spec/SpecGatherer.php]: Removed. I've squashed the commits for nicer history.

Comment (colemanw) [reply Civi/Api4/Service/Spec/SpecGatherer.php]: Perfect. Let's see what the test suite thinks of this.

Comment (semseysandor) [reply Civi/Api4/Service/Spec/SpecGatherer.php]: Looks like test suite doesn't like it...
I believe the reason is a missing `<default>1</default>` on civicrm_acl.is_active.
I've added a quick hack to simulate that and to verify tests pass with that.

Comment (semseysandor) [reply Civi/Api4/Service/Spec/SpecGatherer.php]: Quick hack didn't work so I did the proper way :smile: 
Missing default on `civicrm_acl.is_active` is just _part_ of the reasons... Though there are 2 less failing test cases.

Seems, we need to go through the rest of the failing tests.
I think it's better to divide this PR into smaller parts and extract missing default on `civicrm_acl.is_active` as it's independent of this and needs to fixed anyway.

**Pull request closed without merging**